caddyconfig/httpcaddyfile/builtins.go

```diff
@@ -858,3 +859,30 @@ func parseLogHelper(h Helper, globalLogNames map[string]struct{}) ([]ConfigValue
 	}
 	return configValues, nil
 }
+
+// parseSkipLog parses the skip_log directive. Syntax:
+//
+//     skip_log <matcher>
```

Comment:
Can `skip_log` be used within a `handle` block? (Why is a matcher _always_ required?)

Reply:
I guess yeah you could have it nested within a handle which does have a matcher.

But the way I was thinking of it is why would you skip logging for all requests if you enabled logging? I guess we can loosen that.

Reply:
I think we should loosen the _absolute_ requirement that a matcher is required, because `route` and `handle` blocks that apply to only some of the requests are a very valid way to use `skip_log`.

But yeah I agree, it doesn't make sense to use this without _any_ matcher applied. Just not sure it's worth enforcing it.

Reply:
If you want to fix it up, feel free 😅

Reply:
It just occurred to me while refactoring this (and fixing the merge conflict that I broke, d'oh) that we don't need a directive necessarily:

```
vars http.skip_log true
```

WDYT?

Reply:
That's true but the amount of demand for this I think means we should have first-class support for this rather than hiding it behind a var. Way more discoverable as a directive. Same reason we have `root` as a directive.

Reply:
That's a valid point I guess.